src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/CopiedFieldRvaNode.cs

```diff
@@ -46,23 +48,64 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
             ObjectDataBuilder builder = new ObjectDataBuilder(factory, relocsOnly);
             builder.RequireInitialPointerAlignment();
             builder.AddSymbol(this);
+            builder.EmitBytes(GetRvaData(factory.Target.PointerSize));
+            return builder.ToObjectData();
+        }
 
-            builder.EmitBytes(_field.GetFieldRvaData());
+        private unsafe byte[] GetRvaData(int targetPointerSize)
+        {
+            int size = 0;
+            byte[] result = Array.Empty<byte>();
 
-            return builder.ToObjectData();
+            MetadataReader metadataReader = _module.MetadataReader;
+            BlobReader metadataBlob = new BlobReader(_module.PEReader.GetMetadata().Pointer, _module.PEReader.GetMetadata().Length);
+            metadataBlob.Offset = metadataReader.GetTableMetadataOffset(TableIndex.FieldRva);
+
+            ImmutableArray<byte> memBlock = _module.PEReader.GetSectionData(_rva).GetContent();
+
+            for (int i = 1; i <= metadataReader.GetTableRowCount(TableIndex.FieldRva); i++)
+            {
+                int currentFieldRva = metadataBlob.ReadInt32();
+                short currentFieldRid = metadataBlob.ReadInt16();
+                if (currentFieldRva != _rva)
+                    continue;
+
+                EcmaField field = (EcmaField)_module.GetField(MetadataTokens.FieldDefinitionHandle(currentFieldRid));
+                Debug.Assert(field.HasRva);
+
+                int currentSize = field.FieldType.GetElementSize().AsInt;
+                if (currentSize > size)
+                {
+                    if (currentSize > memBlock.Length)
+                        throw new BadImageFormatException();
+
+                    // We need to handle overlapping fields by reusing blobs based on the rva, and just update
+                    // the size and contents
+                    size = currentSize;
+                    result = new byte[AlignmentHelper.AlignUp(size, targetPointerSize)];
+                    memBlock.CopyTo(0, result, 0, size);
+                }
+            }
+
+            Debug.Assert(size > 0);
+            return result;
         }
 
         protected override string GetName(NodeFactory factory) => this.GetMangledName(factory.NameMangler);
 
         public void AppendMangledName(NameMangler nameMangler, Utf8StringBuilder sb)
         {
             sb.Append(nameMangler.CompilationUnitPrefix);
-            sb.Append($"_FieldRva_{nameMangler.GetMangledFieldName(_field)}");
+            sb.Append($"_FieldRvaData_{_module.Assembly.GetName().Name}_{_rva}");
```

Comment:
Before, we were appending mangled field name, now we don't seem to include the field name anymore. Is it intended?

Reply:
Yes. There can be multiple fields with data at the same RVA. The important thing here is the RVA, not the field itself.